src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -44,7 +44,7 @@ private ClassName(List<String> names) {
       checkArgument(SourceVersion.isName(names.get(i)), "part '%s' is keyword", names.get(i));
     }
     this.names = Util.immutableList(names);
-    this.canonicalName = names.get(0).isEmpty()
```

Comment:
`names.get(0)` can be `null` sometimes. Didn't find the cause, though. But if you remove the check `TypesTest` will throw a NPE.